src/telemetry.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import AdsTelemetryReporter, { TelemetryEventMeasures, TelemetryEventProperties } from '@microsoft/ads-extension-telemetry';
+import { IServerInfo } from 'vscode-mssql';
+import { IConnectionProfile } from './models/interfaces';
+import * as vscode from 'vscode';
+
+const packageJson = vscode.extensions.getExtension('ms-mssql.mssql').packageJSON;
+
+let packageInfo = {
+	name: packageJson.name,
+	version: packageJson.version,
+	aiKey: packageJson.aiKey
+};
+
+export enum TelemetryViews {
+	ObjectExplorer = 'ObjectExplorer',
+	CommandPallet = 'CommandPallet',
+	SqlProjects = 'SqlProjects',
+	QueryEditor = 'QueryEditor',
+	ResultsGrid = 'ResultsGrid',
+	ConnectionPrompt = 'ConnectionPrompt'
+}
+
+export enum TelemetryActions {
+	Scripting = 'Scripting',
+	Refresh = 'Refresh',
+	CreateProject = 'CreateProject',
+	RemoveConnection = 'RemoveConnection',
+	Disconnect = 'Disconnect',
+	NewQuery = 'NewQuery',
+	RunQuery = 'RunQuery',
+	QueryExecutionCompleted = 'QueryExecutionCompleted',
+	ResultPaneAction = 'ResultPaneAction',
+	CreateConnection = 'CreateConnection',
+	ConnectionCreated = 'ConnectionCreated',
```

Comment:
similar as above comments, ConnectionCreated and ConnectionFailed should be a property of the createConnection action.